OrbitClientModel/CaptureSerializerTest.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureData.h"
+#include "FunctionUtils.h"
+#include "OrbitClientModel/CaptureSerializer.h"
+#include "OrbitProcess.h"
+#include "TracepointCustom.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_cat.h"
+#include "capture_data.pb.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using orbit_client_protos::CallstackEvent;
+using orbit_client_protos::CallstackInfo;
+using orbit_client_protos::CaptureInfo;
+using orbit_client_protos::FunctionInfo;
+using orbit_client_protos::FunctionStats;
+using orbit_client_protos::LinuxAddressInfo;
+using ::testing::ElementsAreArray;
+
+TEST(CaptureSerializer, GetCaptureFileName) {
+  CaptureData capture_data;
+
+  time_t timestamp = std::chrono::system_clock::to_time_t(capture_data.capture_start_time());
+  std::string expected_file_name = absl::StrCat("_", OrbitUtils::FormatTime(timestamp), ".orbit");
+  EXPECT_EQ(expected_file_name, capture_serializer::GetCaptureFileName(capture_data));
+}
+
+TEST(CaptureSerializer, IncludeOrbitExtensionInFile) {
+  std::string file_name_with_extension = "process_000.orbit";
+  std::string expected_file_name = file_name_with_extension;
+  capture_serializer::IncludeOrbitExtensionInFile(file_name_with_extension);
+  EXPECT_EQ(expected_file_name, file_name_with_extension);
+
+  std::string file_name_without_extension = "process_000";
+  capture_serializer::IncludeOrbitExtensionInFile(file_name_without_extension);
+  EXPECT_EQ(expected_file_name, file_name_without_extension);
+}
+
+TEST(CaptureSerializer, GenerateCaptureInfoEmpty) {
+  CaptureData capture_data;
+  absl::flat_hash_map<uint64_t, std::string> key_to_string_map;
+
+  CaptureInfo capture_info =
+      capture_serializer::internal::GenerateCaptureInfo(capture_data, key_to_string_map);
+  EXPECT_EQ(0, capture_info.selected_functions_size());
+  EXPECT_EQ(-1, capture_info.process_id());
+  EXPECT_EQ("", capture_info.process_name());
+  EXPECT_EQ(0, capture_info.address_infos_size());
+  EXPECT_EQ(0, capture_info.callstacks_size());
+  EXPECT_EQ(0, capture_info.callstack_events_size());
+  EXPECT_EQ(0, capture_info.key_to_string_size());
+  EXPECT_EQ(0, capture_info.function_stats_size());
+}
+
+TEST(CaptureSerializer, GenerateCaptureInfo) {
+  int32_t process_id = 42;
+  std::string process_name = "p";
+  auto process = std::make_shared<Process>();
```

Comment:
It looks like @antonrohr is removing the Process in the PR. Did you discuss it or am I misunderstand it?

Reply:
I know. Here I need that process. But it can be easily removed as soon Anton is doing a rebase.
I'll get in touch with him.